<launch>
  <!-- node name: all topics will be advertised under this name -->
  <arg name = "node_name"          default = "royale_camera"/>
  <arg name = "rc_tf_x"            default = "0"/>
  <arg name = "rc_tf_y"            default = "0"/>
  <arg name = "rc_tf_z"            default = "0"/>
  <arg name = "rc_tf_yaw"          default = "0"/>
  <arg name = "rc_tf_pitch"        default = "0"/>
  <arg name = "rc_tf_roll"         default = "0"/>
  <arg name = "rc_inv_z"           default = "0"/>
  <arg name = "royale_access_code" default = "Sh!2CBpf"/>
  <arg name = "royale_cam_record"  default = "out.mp4"/>
  <arg name = "royale_cam_id"      default = ""/>
  <arg name = "royale_use_case"    default = ""/>
  
  <!-- static TF publisher -->
  <node pkg = "tf" type = "static_transform_publisher" name = "$(arg node_name)_static_tf"
        args = "$(arg rc_tf_x) $(arg rc_tf_y) $(arg rc_tf_z) $(arg rc_tf_yaw) $(arg rc_tf_pitch) $(arg rc_tf_roll) royale_$(arg node_name)_link royale_camera_$(arg node_name)_optical_frame 100" />

  <!-- nodelet manager -->
  <node pkg = "nodelet" type = "nodelet" name = "$(arg node_name)" args = "manager" output = "screen"/>

  <!-- camera_driver nodelet -->
  <node pkg = "nodelet" type = "nodelet" name = "$(arg node_name)_driver"
        args = "load royale_in_ros/royale_nodelet $(arg node_name)"
        respawn = "true" output = "screen">
    <param name="node_name"          type="str"   value="$(arg node_name)" />
    <param name="rc_inv_z"           type="int"   value="$(arg rc_inv_z)" />
    <param name="royale_cam_id"      type="str"   value="$(arg royale_cam_id)" />
    <param name="royale_cam_record"  type="str"   value="$(arg royale_cam_record)" />
    <param name="royale_access_code" type="str"   value="$(arg royale_access_code)" />
    <param name="royale_use_case"    type="str"   value="$(arg royale_use_case)" />
  </node> 
</launch>
